client/lib/url-search-query-state/index.js

```diff
@@ -0,0 +1,67 @@
+import page from 'page';
+import { Component, useEffect, useState, useCallback } from 'react';
+
+function updateUrlSearchQuery( queryName, queryValue ) {
+	if ( ! window ) {
+		return;
+	}
+	const searchParams = new URLSearchParams( window.location.search );
+	// let historyObject = null;
+	let newQuery = '';
+	if ( queryValue ) {
+		searchParams.set( queryName, queryValue );
+	} else {
+		searchParams.delete( queryName );
+	}
+
+	if ( searchParams.toString() ) {
+		newQuery = '?' + decodeURIComponent( searchParams.toString() );
+	}
+	page( window.location.pathname + newQuery );
+	// Dispatch the custom event.
+	const event = new CustomEvent( 'changeUrlSearchQuery', { details: { queryName, queryValue } } );
```

Comment:
Why do we need the custom event? Can you elaborate? Also, is the `details` property used anywhere?

Reply:
In my testing, the code wasn't always rerendered as expected when the user clicked the button to show or close the dialogue. Also, it helps centralize the visibility of the dialogue to one prop vs having the visibility come from the redux store or the url.

Reply:
The details property is not currently used anywhere but I think it might make sense to have it so that we don't rerender components. This could be used when we have more than one different url query that we are listening for and an issue arises.

Reply:
but will remove this for now.